tidy3d/components/time_modulation.py

```diff
@@ -0,0 +1,207 @@
+""" Defines time modulation to the medium"""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Union
+from math import isclose
+
+import pydantic.v1 as pd
+import numpy as np
+
+from .base import Tidy3dBaseModel, cached_property
+from .types import InterpMethod
+from .source import AbstractTimeDependence
+from .data.data_array import SpatialDataArray
+from ..exceptions import ValidationError
+from ..constants import HERTZ
+
+
+class AbstractTimeModulation(AbstractTimeDependence):
+    """Base class for modulation in time.
+
+    Note
+    ----
+    This class describes the time dependence part of the separable space-time modulation type
+    as shown below,
+    .. math::
+
+        amp(r, t) = \\Re[amp\\_time(t) \\cdot amp\\_space(r)]
+
+    """
+
+    @cached_property
+    @abstractmethod
+    def max_modulation(self) -> float:
+        """Estimated maximal modulation amplitude."""
+
+
+class ContinuousWaveTimeModulation(AbstractTimeDependence):
+    """The modulation in time is in continuous oscillation.
```

Comment:
Class describing modulation with a harmonic time dependence? or something like this maybe? or is "Continuous oscillation" well understood?